src/collectors/proc.plugin/sys_class_power_supply.c

```diff
@@ -5,20 +5,35 @@
 #define PLUGIN_PROC_MODULE_POWER_SUPPLY_NAME "/sys/class/power_supply"
 #define PROP_VALUE_LENGTH_MAX 30
 
-const char *ps_property_names[]  = {        "charge",         "energy",              "voltage"};
-const char *ps_property_titles[] = {"Battery charge", "Battery energy", "Power supply voltage"};
-const char *ps_property_units[]  = {            "Ah",             "Wh",                    "V"};
+const char *ps_property_names[]  = {"capacity", "power", "charge", "energy", "voltage"};
+const char *ps_property_titles[] = {"Battery capacity", "Battery power", "Battery charge", "Battery energy", "Power supply voltage"};
+const char *ps_property_units[]  = {"percentage", "W", "Ah", "Wh", "V"};
 
 const long ps_property_priorities[] = {
+    NETDATA_CHART_PRIO_POWER_SUPPLY_CAPACITY,
+    NETDATA_CHART_PRIO_POWER_SUPPLY_POWER,
     NETDATA_CHART_PRIO_POWER_SUPPLY_CHARGE,
     NETDATA_CHART_PRIO_POWER_SUPPLY_ENERGY,
     NETDATA_CHART_PRIO_POWER_SUPPLY_VOLTAGE
 };
 
+enum {
+    PS_PROP_CAPACITY,
+    PS_PROP_POWER,
+    PS_PROP_CHARGE,
+    PS_PROP_ENERGY,
+    PS_PROP_VOLTAGE,
+    PS_PROP_END
+};
+
+const unsigned long ps_property_divisors[] = {1, 1000000, 1000000, 1000000, 1000000 };
 
-const char *ps_property_dim_names[] = {"empty_design", "empty", "now", "full", "full_design",
-                                       "empty_design", "empty", "now", "full", "full_design",
-                                         "min_design",   "min", "now",  "max",  "max_design"};
+const char *ps_property_dim_names[] = {
```

Comment:
Hello,

I've tested your code, and it seems to be functional. However, upon reviewing it, I've come across some changes that I'm not entirely convinced about. Let me explain:

We're introducing two new rows here, along with a NULL at the end of each, but the "table" doesn't maintain a consistent number of columns for each line. This necessitates changes to the subsequent code.

I believe there's potential to simplify the changes you're proposing. My suggestion is to maintain the previous format of "ps_property_dim_names." Instead, consider creating a function within the loop that handles the necessary arguments for running the algorithm.

Following these adjustments, before entering the loop, you could execute calls for the lines that don't require five values, and then proceed with the loop for the remaining code.

I bring this up because I believe that introducing a "table" with varying patterns could potentially confuse some developers.  :thinking: